src/core/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/side-navigation.css';
+import { CommonProps } from '../utils/props';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import { IconButton } from '../Buttons';
+
+export type SidebarProps = {
+  /**
+   * Buttons shown in the top portion sidebar.
+   * Recommended to use `SidenavButton`s.
+   */
+  mainItems: React.ReactNode[];
+  /**
+   * Buttons shown at the bottom of the sidebar.
+   */
+  secondaryItems?: React.ReactNode[];
+  /**
+   * Control the visibility of "expander" icon button.
+   * @default 'top'
+   */
+  expanderVisibility?: 'top' | 'bottom' | 'hidden';
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Describe me here!
+ * @example
+ * Example usages go here!
+ */
+export const Sidebar = (props: SidebarProps) => {
+  const {
+    mainItems,
+    secondaryItems,
+    expanderVisibility = 'top',
+    className,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  const ExpandButton = (
+    <IconButton
+      className='iui-sidenav-button iui-expand'
+      onClick={() => setIsExpanded((expanded) => !expanded)}
+    >
+      <SvgChevronRight />
+    </IconButton>
+  );
+
+  return (
+    <div
+      className={cx(
+        'iui-side-navigation',
+        {
+          'iui-expanded': isExpanded,
+          'iui-collapsed': !isExpanded,
+        },
+        className,
+      )}
+      {...rest}
+    >
+      {expanderVisibility === 'top' && ExpandButton}
+      <div className='iui-sidenav-content'>
+        <div className='iui-top'>
+          {mainItems.map((sidebarButton) =>
+            React.cloneElement(sidebarButton as JSX.Element, {
+              showTooltip: !isExpanded,
+            }),
+          )}
```

Comment:
This is the best I could come up with after some brainstorming.

It still needs some work to validate that it's a `SidebarButton` element.

Reply:
What about wrapping with a `Tooltip` here and switching between just a `button` and then wrapped with `Tooltip`?

Reply:
I tried this but the tooltip never shows up. And it's not a problem with `sidebarButton.props.children` because I'm able to get the tooltip to show if I wrap just a static span instead of `sidebarButton`.

```jsx
{mainItems.map((sidebarButton: JSX.Element) =>
  !isExpanded ? (
    <Tooltip content={sidebarButton.props.children} placement='right'>
      {sidebarButton}
      // also tried these two:
      // React.cloneElement(sidebarButton)
      // (() => sidebarButton())()
    </Tooltip>
  ) : (
    sidebarButton
  ),
)}
```

Reply:
Update: it works now. Just had to forward the ref in `SidebarButton` 😆 

Reason: https://github.com/atomiks/tippyjs-react#component-children

Reply:
You had the same issue that Basanta told us about 😀 Maybe add this link to out `Popover` or even to all components that are using `Popover`.

Reply:
Popover already points to tippyjs though 🤔